cohortextractor2/dsl.py

```diff
@@ -243,7 +243,7 @@ def not_null_patient_series(patient_series: PatientSeries) -> PatientSeries:
 
 
 def categorise(
-    mapping: dict[Expression, PatientSeries], default: Expression | None = None
+    mapping: Mapping[Expression, PatientSeries], default: Expression | None = None
```

Comment:
What does `Mapping` give us?

Reply:
`Mapping` is covariant in the value type, `dict` isn't. I actually _don't_ need that for this PR but used it in the spike this is based on when we need to pass in different types of `PatientSeries`.

Actually I think we might need a bit of rework for the typing of categories, so I'll pull that out now and we can reintroduce it later if necessary.